### PR TITLE
[project-pineapples < T1020] Add machine manager prototype

### PR DESCRIPTION
**Pull request by spacejam**
[master < Epic] PR
- [ ] Check, and update documentation if necessary
- [ ] Update [changelog](https://docs.memgraph.com/memgraph/changelog)
- [ ] Write E2E tests
- [ ] Compare the [benchmarking results](https://bench-graph.memgraph.com/) between the master branch and the Epic branch
- [ ] Provide the full content or a guide for the final git message

[master < Task] PR
- [ ] Check, and update documentation if necessary
- [ ] Update [changelog](https://docs.memgraph.com/memgraph/changelog)
- [ ] Provide the full content or a guide for the final git message

### CODE REVIEW

**Review by 42jeremy — APPROVED** [review]

File: src/coordinator/coordinator.cpp [hunk 1]
```diff
@@ -93,9 +93,6 @@ CoordinatorWriteResponses Coordinator::ApplyWrite(
     DeregisterStorageEngineRequest && /* register_storage_engine_request */) {
   DeregisterStorageEngineResponse res{};
   // TODO
```
Comment (gvolfing) [inline src/coordinator/coordinator.cpp:95]: Is there an aim of this TODO, or ist just some left over?

File: src/coordinator/coordinator.hpp [hunk 2]
```diff
@@ -177,11 +177,11 @@ class Coordinator {
 
   /// This adds the provided storage engine to the standby storage engine pool,
   /// which can be used to rebalance storage over time.
-  CoordinatorWriteResponses ApplyWrite(RegisterStorageEngineRequest && /* register_storage_engine_request */);
+  static CoordinatorWriteResponses ApplyWrite(RegisterStorageEngineRequest && /* register_storage_engine_request */);
```
Comment (gvolfing) [inline src/coordinator/coordinator.hpp:180]: I am not sure if these should be static. Shouldn't we just use the clang-tidy suppression comments?

File: src/machine_manager/machine_config.hpp [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  size_t replication_factor;
+  std::vector<CompoundKey> split_points;
+};
+
+struct MachineConfig {
+  std::vector<InitialLabelSpace> initial_label_spaces;
+  std::vector<Address> coordinator_addresses;
+  bool is_storage;
+  bool is_coordinator;
+  bool is_query_engine;
```
Comment (gvolfing) [inline src/machine_manager/machine_config.hpp:37]: Enumerations may be somewhat nicer.

**Review by gvolfing — APPROVED** [review]

File: src/coordinator/coordinator.hpp (hunk 2 above)
Comment (spacejam) [reply src/coordinator/coordinator.hpp:180]: yeah, but it's trivial for now to just get it to be quiet with static

File: src/machine_manager/machine_config.hpp (hunk 3 above)
Comment (gvolfing) [reply src/machine_manager/machine_config.hpp:37]: Oh I see that makes sense.

**Pull request merged**